clockwork.go

```diff
@@ -227,21 +213,45 @@ func (fc *fakeClock) Advance(d time.Duration) {
 }
 
 // BlockUntil blocks until the fakeClock has the given number of waiters.
+//
+// Deprecated: Prefer BlockUntilContext.
```

Comment:
I wonder if we should deprecate this. I can see why not having a deadline can cause trouble during test runs, but at the same time the very reason why a new function is introduced for that is to keep backwards compatibility.

Let's not deprecate it just yet. Maybe leave a comment that we may deprecate it in the future.

Thanks!

Reply:
Done.

I used the term "Deprecation warning", let me know if you prefer different version. I have no strong feelings other than we should slowly migrate to a version that supports contexts.